semantic/src/main/kotlin/tools/samt/semantic/checks/ValidTypeExpressionCheck.kt

```diff
@@ -10,18 +10,28 @@ internal class ValidTypeExpressionCheck(private val diagnostics: DiagnosticConso
             when (it) {
                 is RecordDeclarationNode -> it.fields.forEach { field -> checkTypeExpression(field.type) }
                 is TypeAliasNode -> checkTypeExpression(it.type)
+                is ServiceDeclarationNode -> it.operations.forEach { operation -> checkOperationType(operation) }
                 is TypeImportNode,
                 is WildcardImportNode,
                 is EnumDeclarationNode,
                 is ConsumerDeclarationNode,
                 is PackageDeclarationNode,
                 is ProviderDeclarationNode,
-                is ServiceDeclarationNode,
                 -> Unit
             }
         }
     }
 
+    private fun checkOperationType(operation: OperationNode) {
```

Comment:
I think we're missing a validation step here to make sure operation parameter names are unique.

Reply:
Has been added